libobs/obs-hotkey.c

```diff
@@ -1194,8 +1194,11 @@ static inline bool modifiers_match(obs_hotkey_binding_t *binding,
 				   uint32_t modifiers_, bool strict_modifiers)
 {
 	uint32_t modifiers = binding->key.modifiers;
-	return !modifiers ||
-	       (!strict_modifiers && (modifiers & modifiers_) == modifiers) ||
+	// If not strict, require only the modifiers (including NO modifiers)
+	//   requested by the hotkey definition. (legacy behavior)
+	// If strict, the current modifiers (including NO modifiers) must
+	//   match the hotkey definition. ("shortcut" behavior)
+	return (!strict_modifiers && (modifiers & modifiers_) == modifiers) ||
 	       (strict_modifiers && modifiers == modifiers_);
```

Comment:
```suggestion
	if (!strict_modifiers)
		return !modifiers || (modifiers & modifiers_) == modifiers;
	else
		return modifiers == modifiers_;
```
Rather than adding comments, I'd like to recommend to use an `if`-`else` statement.

Reply:
To reviewers,
`!modifiers || (modifiers & modifiers_) == modifiers` is equivalent to `(modifiers & modifiers_) == modifiers` so that it is ok to remove `!modifiers ||`.

Reply:
I think squash/rebase is preferable in this project to keep the history of the master branch as simple as possible.

Reply:
if/then incorporated, squash/rebase done